### DISCUSSION TITLE
Active Directory sudo

### DISCUSSION THREAD

**Discussion by luik8004** [ltsp/ltsp#886] [General]
When installing Ubuntu 22.04.3 Desktop, there is an option to join Active Directory (AD). After entering the login and password, you are already in the domain. After installing Ubuntu, you can log in with AD users, and there is also an option to log in using the local administrator account. After installing LTSP, creating an image, and booting from the client, logging in as an AD user is possible, and a user directory is created in /home. However, on the client, there is no option to log in with the local administrator account.
Based on this, when the client boots the LTSP image and logs in with an AD account, attempting to execute a sudo command prompts for the administrator password. If the password is entered, it reports that the password is incorrect, even though it works on the server. How can one use the sudo command for an AD user?

**Comment by macroendrix** [comment]
You need to create a group in Active Directory and add the users who are supposed to be able to run sudo, then add that group to /etc/sudoers. For example, if groups names in Active Directory are "Desktop Admins" and "Domain Admins", then in /etc/sudoers you must specify:

%desktop\ admins ALL=(ALL:ALL) ALL
%domain\ admins ALL=(ALL:ALL) ALL

That's all, greetings.

**Reply by luik8004** [reply]
I created a group named "LTSP Admins" in Active Directory. The group's scope is "Global," and its type is "Security." I added the user "bb1" to the LTSP Admins group. Next, on the LTSP server, I execute the command "sudo visudo" to edit the "/etc/sudoers" file. After the lines:
# Cmnd alias specification

# User privilege specification
root    ALL=(ALL:ALL) ALL

I added the following entry:

%LTSP\ Admins ALL=(ALL:ALL) ALL

I save and exit. Then I run the command "ltsp image /" followed by "ltsp initrd." After that, I boot from the client, log in as the user "[bb1@example.com](mailto:bb1@example.com)." I open the terminal and try to execute a command, for example, "sudo cat /etc/sudoers." It prompts me for the password for "[bb1@example.com](mailto:bb1@example.com)"; I enter it, and I receive the message that "[bb1@example.com](mailto:bb1@example.com) is not in the sudoers file. This incident will be reported."

What am I doing wrong? :)

**Comment by macroendrix** [comment]
The only difference I notice is the capitalization of the group name in "/etc/sudoers". Did you tried with:

%ltsp\ admins ALL=(ALL:ALL) ALL

Greetings

**Comment by nogajun** [comment]
Please enclose the group name in double quotes. I have been able to do this successfully.

```conf
"%teacher"  ALL=(ALL) ALL
"%student"  ALL=(ALL) NOPASSWD:/usr/bin/docker
```

To configure sudo, create a file under /etc/sudoers.d/ with permission 0440.
